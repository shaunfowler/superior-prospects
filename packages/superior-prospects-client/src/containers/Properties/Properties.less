.propertiesView {
  .location-tabs {
    border-bottom: 1px solid #eee;
    background: #fafafa;
  }

  .location-summary {
    padding: 15px;
  }

  .location-select {
    margin-top: 20px;
  }

  .location-toolbar {
    display: flex;
    justify-content: flex-end;
    margin-bottom: 15px;
  }
}
