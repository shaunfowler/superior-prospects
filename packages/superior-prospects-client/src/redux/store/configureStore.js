import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import createHistory from "history/createBrowserHistory";
import { routerMiddleware } from "react-router-redux";
import rootReducer from "../reducers";

export const history = createHistory();

const initialState = {};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const enhancer = composeEnhancers(
  applyMiddleware(
    reduxImmutableStateInvariant(),
    thunk,
    routerMiddleware(history),
  )
);

const store = createStore(rootReducer, initialState, enhancer);

export default store;
