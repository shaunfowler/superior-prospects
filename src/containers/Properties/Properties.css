.propertiesView {
  flex: 1;
  padding: 20px 0;
}

.propertiesView .location {
  margin-bottom: 50px;
}
