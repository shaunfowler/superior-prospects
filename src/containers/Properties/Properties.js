import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropertyItem from "../../components/PropertyItem";

const renderPropertiesForLocationId = (properties, locationId) => (
  <div>
    {properties &&
      properties
        .filter(p => p.locationRefId === locationId)
        .map(p => (
          <PropertyItem
            key={p._id}
            id={p._id}
            name={p.name}
            safeName={p.safeName}
            description={p.description}
          />
        ))}
  </div>
);

class Properties extends Component {
  componentWillMount() {
    this.props.queryLocations();
    this.props.queryProperties();
  }

  render() {
    const locations = this.props.locations.list;
    const properties = this.props.properties.list;
    return (
      <div className="propertiesView container">
        <h1 className="title">Properties</h1>
        {locations &&
          locations.map(l => (
            <div key={l._id} className="location">
              <div className="is-size-4 has-text-weight-bold">{l.name}</div>
              <div className="is-text-grey">{l.body}</div>
              {renderPropertiesForLocationId(properties, l._id)}
            </div>
          ))}
      </div>
    );
  }
}

export default Properties;
