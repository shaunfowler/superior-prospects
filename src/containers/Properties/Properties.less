.propertiesView {
  .location-tabs {
    border-bottom: 1px solid #eee;
    background: #fafafa;
  }

  .location-summary {
    padding: 15px;
  }

  .location-select {
    margin-top: 20px;
  }
}
