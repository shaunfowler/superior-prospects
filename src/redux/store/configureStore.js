import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import createHistory from "history/createBrowserHistory";
import { routerMiddleware } from "react-router-redux";
import rootReducer from "../reducers";
import routeTrackerMiddleware from "../middleware/routeTrackerMiddleware";

export const history = createHistory();

const initialState = {};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const enhancer = composeEnhancers(
  applyMiddleware(
    reduxImmutableStateInvariant(),
    thunk,
    routerMiddleware(history),
    routeTrackerMiddleware
  )
);

const store = createStore(rootReducer, initialState, enhancer);

export default store;
