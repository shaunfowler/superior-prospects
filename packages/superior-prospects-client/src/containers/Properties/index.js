import { connect } from "react-redux";
import {
  queryLocations,
  createLocation,
  updateLocation
} from "../../redux/actions/locationsActions";
import {
  queryProperties,
  createProperty
} from "../../redux/actions/propertiesActions";
import Properties from "./Properties";
import "./Properties.less";

const alphabeticalSortPredicate = (a, b) =>
  a.name !== b.name ? (a.name < b.name ? -1 : 1) : 0;

const mapStateToProps = ({ locations, properties, user }) => {
  return {
    locations: [...locations.list].sort(alphabeticalSortPredicate),
    properties: [...properties.list].sort(alphabeticalSortPredicate),
    isAuthenticated: user.isAuthenticated
  };
};

const mapDispatchToProps = dispatch => {
  return {
    queryLocations: () => dispatch(queryLocations()),
    queryProperties: () => dispatch(queryProperties()),
    createLocation: entity => dispatch(createLocation(entity)),
    createProperty: entity => dispatch(createProperty(entity)),
    updateLocation: entity => dispatch(updateLocation(entity))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Properties);
