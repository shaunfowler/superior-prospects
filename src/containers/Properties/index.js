import { connect } from "react-redux";
import {
  queryLocations,
  createLocation,
  updateLocation
} from "../../redux/actions/locationsActions";
import {
  queryProperties,
  createProperty
} from "../../redux/actions/propertiesActions";
import Properties from "./Properties";
import "./Properties.less";

const mapStateToProps = ({ locations, properties, user }) => {
  return {
    locations,
    properties,
    isAuthenticated: user.isAuthenticated
  };
};

const mapDispatchToProps = dispatch => {
  return {
    queryLocations: () => dispatch(queryLocations()),
    queryProperties: () => dispatch(queryProperties()),
    createLocation: entity => dispatch(createLocation(entity)),
    createProperty: entity => dispatch(createProperty(entity)),
    updateLocation: entity => dispatch(updateLocation(entity))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Properties);
