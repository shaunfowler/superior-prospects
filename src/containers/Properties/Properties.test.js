import React from "react";
import ReactDOM from "react-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Properties from "./Properties";

const mockStore = configureStore([thunk]);

it("renders without crashing", () => {
  const initialState = {
    locations: {
      loading: false,
      list: []
    },
    properties: {
      loading: false,
      list: []
    }
  };
  const store = mockStore(initialState);
  const div = document.createElement("div");
  ReactDOM.render(<Properties store={store} />, div);
});
