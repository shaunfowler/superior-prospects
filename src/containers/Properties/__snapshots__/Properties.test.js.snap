// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties container renders when authenticated 1`] = `
<div
  className="propertiesView"
>
  <WithStyles(Dialog)
    aria-labelledby="form-dialog-title"
    open={false}
  >
    <WithStyles(DialogTitle)
      id="form-dialog-title"
    >
      Add Property
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <TextField
        autoFocus={true}
        fullWidth={true}
        label="Location Name"
        onChange={[Function]}
        required={false}
        select={false}
        type="text"
        value=""
      />
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithStyles(Button)>
      <WithStyles(Button)
        color="primary"
        disabled={true}
        onClick={[Function]}
      >
        Add
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
  <WithStyles(Dialog)
    aria-labelledby="form-dialog-title"
    open={false}
  >
    <WithStyles(DialogTitle)
      id="form-dialog-title"
    >
      Add Property
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <TextField
        autoFocus={true}
        fullWidth={true}
        label="Property Name"
        onChange={[Function]}
        required={false}
        select={false}
        type="text"
        value=""
      />
      <br />
      <br />
      <WithStyles(FormControl)>
        <WithStyles(InputLabel)
          htmlFor="location"
        >
          Location
        </WithStyles(InputLabel)>
        <WithStyles(Select)
          fullWidth={true}
          inputProps={
            Object {
              "id": "location",
              "name": "location",
            }
          }
          onChange={[Function]}
          value={null}
        />
      </WithStyles(FormControl)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithStyles(Button)>
      <WithStyles(Button)
        color="primary"
        disabled={true}
        onClick={[Function]}
      >
        Add
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
  <div
    className="paper__title"
  >
    <WithStyles(Typography)
      variant="title"
    >
      Properties
    </WithStyles(Typography)>
    <div>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Add Location
      </WithStyles(Button)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Add Property
      </WithStyles(Button)>
    </div>
  </div>
  <WithStyles(Tabs)
    className="location-tabs"
    fullWidth={true}
    indicatorColor="primary"
    onChange={[Function]}
    textColor="primary"
    value={0}
  >
    <WithStyles(Tab)
      label="All Locations"
    />
  </WithStyles(Tabs)>
  <WithStyles(List) />
</div>
`;

exports[`Properties container renders when not authenticated 1`] = `
<div
  className="propertiesView"
>
  <div
    className="paper__title"
  >
    <WithStyles(Typography)
      variant="title"
    >
      Properties
    </WithStyles(Typography)>
  </div>
  <WithStyles(Tabs)
    className="location-tabs"
    fullWidth={true}
    indicatorColor="primary"
    onChange={[Function]}
    textColor="primary"
    value={0}
  >
    <WithStyles(Tab)
      label="All Locations"
    />
  </WithStyles(Tabs)>
  <WithStyles(List) />
</div>
`;

exports[`Properties container renders with properties and locations 1`] = `
<div
  className="propertiesView"
>
  <div
    className="paper__title"
  >
    <WithStyles(Typography)
      variant="title"
    >
      Properties
    </WithStyles(Typography)>
  </div>
  <WithStyles(Tabs)
    className="location-tabs"
    fullWidth={true}
    indicatorColor="primary"
    onChange={[Function]}
    textColor="primary"
    value={0}
  >
    <WithStyles(Tab)
      label="All Locations"
    />
    <WithStyles(Tab)
      key="abc"
      label="loc 1"
    />
  </WithStyles(Tabs)>
  <WithStyles(List)>
    <PropertyItem
      id="123"
      key="123"
      locationName="loc 1"
      name="prop 1"
    />
  </WithStyles(List)>
</div>
`;
